.github/workflows/deploy-prerelease.yml

```
@@ -0,0 +1,224 @@
+
+      - name: Get environment name
+        id: get-env-name
+        run: echo "full-name=$(yq '.name' < ${{env.ENV_NAME}})-${{env.VERSION}}" >> $GITHUB_OUTPUT
+
+      # Search for the latest open PR (not a draft) in the replace_landsurface and era5_grib_parallel repos with
+      # changes in specific files. If a PR is found, download the artifact from the PR latest 
+      # (successful) workflow run. If a PR is not found, download the lates released version.
+      - name: Get latest dependencies
+        id: get-dependencies
+        run: |
+          download_latest_release() {
+            gh release download $(gh release view --repo ${{ github.repository_owner }}/$1 --json tagName --jq '.tagName') \
+            --repo ${{ github.repository_owner }}/$1 --pattern *.tar.bz2 -D ${1}${{env.CHANNEL_SUFFIX}}/${{env.ARCH}} && \
+            echo "${1}: Successfully downloaded latest release: \"$(basename ${1}${{env.CHANNEL_SUFFIX}}/${{env.ARCH}}/*.tar.bz2)\"."
+          }
+          download_run_artifact() {
+            (gh run download $run_id --repo ${{ github.repository_owner }}/$1 -D ${1}${{env.CHANNEL_SUFFIX}}/${{env.ARCH}} && \
+            echo "${1}: Successfully downloaded artifact \"$(basename ${1}${{env.CHANNEL_SUFFIX}}/${{env.ARCH}}/artifact/*.tar.bz2)\"." && \
+            mv ${1}${{env.CHANNEL_SUFFIX}}/${{env.ARCH}}/artifact/*.tar.bz2 ${1}${{env.CHANNEL_SUFFIX}}/${{env.ARCH}}) || \
+            (echo "${1}: No valid artifact found. Dependency will be installed from the latest release.";\
+            download_latest_release ${1})
+          }
+
+
+          for dependency_repo in ${{env.DEPENDENCY_REPOS}}; do
+            echo "Getting latest \"${dependency_repo}\" dependency ..."
+    
+            # Get sha and number of latest open PR that changes either of the following files:
+            # [src/**, setup.py, pyproject.toml, .conda/**]
+            pr=$(gh pr list --repo ${{ github.repository_owner }}/${dependency_repo} --state open --draft=False \
+            --json headRefOid,files,number,url --jq '.[] | select(.files[].path | 
+            (startswith("src/") or (. == "setup.py") or (. == "pyproject.toml") or (startswith(".conda/"))))' \
+            | head -n 1)
+              
+            # if a PR is found, find the ID of the latest successful workflow run
+            if [[ -n "$pr" ]]; then
+              pr_sha=$(jq -r '.headRefOid' <<< $pr)
+              pr_number=$(jq '.number' <<< $pr)
+              pr_numbers+=($pr_number)
+              pr_url=$(jq -r '.url' <<< $pr)
+              echo "${dependency_repo}: Found PR #${pr_number}."
+              echo "  PR url: $pr_url"
+              echo "  Commit ref: $pr_sha"
+    
+              run=$(gh run list --repo ${{ github.repository_owner }}/${dependency_repo} --json databaseId,headSha,event,status,url \
+              --jq ".[] | select(.event == \"pull_request\" and .status == \"completed\" and .headSha == \"$pr_sha\")"\
+              | head -n 1)
```

Regarding injecting the `$pr_sha`. Potentially another way to do this is to break the commands up into getting the json and parsing it with `jq`, something like this (I don't know if I've got all the syntax right):
```suggestion
              run_json=$(gh run list --repo ${{ github.repository_owner }}/${dependency_repo} --json databaseId,headSha,event,status,url)
              run=$(jq \
                  --arg pr_sha "$pr_sha" \
                  '.[] | select(.event == "pull_request" and .status == "completed" and .headSha == "$pr_sha")" \
                  <<< $pr_json \
                | head -n 1
              )
```

---

I think one single command should be fine. I formatted as above to make it clearer.